crates/bevy_ui/src/focus.rs

```diff
@@ -111,16 +116,36 @@ pub fn ui_focus_system(
     let cursor_position = camera
         .iter()
         .filter(|(_, camera_ui)| !is_ui_disabled(*camera_ui))
-        .filter_map(|(camera, _)| {
-            if let RenderTarget::Window(window_id) = camera.target {
-                Some(window_id)
-            } else {
-                None
+        .find_map(|(camera, _)| {
+            match &camera.target {
+                RenderTarget::Window(window_id) => windows.get(*window_id).and_then(|window| {
+                    if window.is_focused() {
+                        window.cursor_position()
+                    } else {
+                        None
+                    }
```

Comment:
This could be simplified to
```suggestion
                    window.is_focused().then(|| window.cursor_position())
```

Reply:
Good catch, I've done these changes, though I did have to call `.flatten()` at the end.